### ISSUE TITLE
mIOU is not expected

### ISSUE THREAD

**Issue by h1063135843** [SwinTransformer/Swin-Transformer-Semantic-Segmentation#17]
`python tools/train.py configs/swin/upernet_swin_tiny_patch4_window7_512x512_160k_ade20k.py --options model.pretrained=checkpoints/swin_tiny_patch4_window7_224.pth model.backbone.use_checkpoint=True --work-dir=./`
I trained the model with above cmd on Windows. mIOU is 30.91 much less than 44.51 as expected. The only change I made was change SyncBN to BN in order to train with single gpu. Is it normal that the mIou differs so much?

**Comment by impiga** [comment]
Hi, @h1063135843 

We use 2 images/gpu x 8 gpus by default. To train with single gpu, longer training may be needed.
Also, you need to tune the learning rate parameter as the batch size differs.

**Comment by infinite013** [comment]
> Hi, @impiga
> thank you. It took one day for me to train 160k schedule. Does it mean that I should train 8 days to arrive the performance as expected. It also accounts for the another problem I met. Training with two gpu even spends more time than single gpu. I thought it might need half the time before.


Hi，have you solved this problem？
I trained the model on 4rtx2080Ti ,mIOU is 21.56

**Comment by h1063135843** [comment]
It seems to be a different problem. Do you use the pretrained model on ImageNet?



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年5月20日(星期四) 下午3:15
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [SwinTransformer/Swin-Transformer-Semantic-Segmentation] mIOU is not expected (#17)





  
Hi, @impiga
 thank you. It took one day for me to train 160k schedule. Does it mean that I should train 8 days to arrive the performance as expected. It also accounts for the another problem I met. Training with two gpu even spends more time than single gpu. I thought it might need half the time before.
  
Hi，have you solved this problem？
 I trained the model on 4rtx2080Ti ,mIOU is 21.56

**Comment by infinite013** [comment]
> It seems to be a different problem. Do you use the pretrained model on ImageNet?
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: ***@***.***&gt;; 发送时间: 2021年5月20日(星期四) 下午3:15 收件人: ***@***.***&gt;; 抄送: ***@***.***&gt;; ***@***.***&gt;; 主题: Re: [SwinTransformer/Swin-Transformer-Semantic-Segmentation] mIOU is not expected (#17) Hi, @impiga thank you. It took one day for me to train 160k schedule. Does it mean that I should train 8 days to arrive the performance as expected. It also accounts for the another problem I met. Training with two gpu even spends more time than single gpu. I thought it might need half the time before. Hi，have you solved this problem？ I trained the model on 4rtx2080Ti ,mIOU is 21.56 — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

yes.The cmd i used is 
CUDA_VISIBLE_DEVICES=0,1,2,3 tools/dist_train.sh configs/swin/upernet_swin_tiny_patch4_window7_512x512_160k_ade20k.py 4 
    --options model.pretrained="/home/song/Swin-Transformer-Semantic-Segmentation/pretrained/swin_tiny_patch4_
                                                                                                                                                                                 window7_224.pth"

but i found that the loss didn't drop

**Comment by qizhuli** [comment]
@impiga Hi there! I was trying to train your model on 4 cards, by increasing the number of samples per card to 4. The total batch size = 4x4 = 8x2 = 16, which is the same as your original setting. Everything else is kept the same. Do you think that would be equivalent to your 8 card experiment?

More generally, do you expect the linear scaling rule presented in [this report](https://arxiv.org/pdf/1706.02677.pdf) to hold true for Swin experiments which use AdamW instead of SGD?
